### PR TITLE
Add a handler for LRU evictions

### PR DESCRIPTION
**Pull request by shajith**
cc: @vcekov

### CODE REVIEW

**Comment by kruppel** [comment]
What's the story behind this change?

**Comment by shajith** [comment]
I want a way to track identity map evictions in production. Might do something about the limits.

File: src/identity_map.js [hunk 1]
```diff
@@ -1,8 +1,15 @@
 /*globals Ember, LRUCache*/
 
 (function() {
-  Ember.Resource.IdentityMap = function(limit) {
+  Ember.Resource.IdentityMap = function(limit, evictionHandler) {
     this.cache = new LRUCache(limit || Ember.Resource.IdentityMap.DEFAULT_IDENTITY_MAP_LIMIT);
+    this.evictionHandler = evictionHandler || function() {};
+    var map = this;
+    this.cache.shift = function() {
```
Comment (vcekov) [inline src/identity_map.js]: While this is fine, the more defensive approach usually is like this:

```js
var origShift = this.cache.shift;

this.cache.shift = function() {
  map.evictionHandler(entry);
  return origShift.apply(this, arguments);
};
```

**Comment by vcekov** [comment]
+1 with comment

File: src/identity_map.js (hunk 1 above)
Comment (vcekov) [inline src/identity_map.js]:
```js
var origShift = this.cache.shift;

this.cache.shift = function() {
  var entry = origShift.apply(this, arguments);
  map.evictionHandler(entry);
  return entry;
};
```

**Pull request merged**